meson.build

```diff
@@ -0,0 +1,117 @@
+py3_version = py3.language_version()
+if py3_version.version_compare('< 3.6')
+  error('Invalid python version!?')
+endif
+
+GR_ENABLE_PYTHON = false
+if (get_option('enable_python'))
+python3_dep = dependency('python3', required : get_option('enable_python'))
+python3_embed_dep = dependency('python3-embed', required : get_option('enable_python'))
+pybind11_dep = dependency('pybind11', required : get_option('enable_python'))
+incdir_numpy = run_command(py3,
+  ['-c', 'import os; os.chdir(".."); import numpy; print(numpy.get_include())'],
+  check : true
+).stdout().strip()
+
+GR_ENABLE_PYTHON = true
+endif
+
+py3 = import('python').find_installation('python3')
+json_dep = dependency('nlohmann_json')
+
+enable_picoscope = get_option('enable_picoscope')
+
+if (get_option('enable_testing'))
+    TEST_ENV = environment()
+    TEST_ENV.prepend('LD_LIBRARY_PATH',
+      join_paths( meson.build_root(),'blocklib','digitizers',get_option('libdir')),
+    )
+    TEST_ENV.prepend('PYTHONPATH', join_paths(meson.build_root(),'blocklib/digitizers/python'))
+
+    if (enable_picoscope)
+      TEST_ENV.prepend('LD_LIBRARY_PATH',
+        join_paths( meson.build_root(),'blocklib','picoscope4000a',get_option('libdir')),
+      )
+      TEST_ENV.prepend('PYTHONPATH', join_paths(meson.build_root(),'blocklib/picoscope4000a/python'))
+    endif
+endif
+
+prefix = get_option('prefix')
+# Escape spaces
+prefix_parts = prefix.split(' ')
+prefix = '\ '.join(prefix_parts)
+
+cuda_available = add_languages('cuda', required : false)
+IMPLEMENT_CUDA = cuda_available and get_option('enable_cuda')
+IMPLEMENT_CPU = true
+
+###############################################################################
+# For OOT modules we need script path from gnuradio, so let's look for
+# gnuradio-runtime already here
+###############################################################################
+gnuradio_gr_dep = dependency('gnuradio-runtime')
+###############################################################################
+# IF Out of Tree Module
+# The script dir is part of the gnuradio installation, not of the oot tree,
+# so we have to find the install prefix of gnuradio
+# Even if we build an In-Tree module, it's the correct location
+###############################################################################
+share_dir = join_paths(gnuradio_gr_dep.get_pkgconfig_variable('prefix'),'share')
+SCRIPTS_DIR = join_paths(share_dir, 'utils','blockbuilder','scripts' )
+###############################################################################
+# Else In-Tree module
+###############################################################################
+# share_dir = meson.project_source_root()
+
+###############################################################################
+# Dependencies
+###############################################################################
+volk_dep = dependency('volk', version : '>=2.2')
+fmt_dep = dependency('fmt', method: 'cmake', modules: ['fmt::fmt'])
```

Comment:
You should not need to specify `method: 'cmake'` here as fmt can be detected correctly via both pkg-config and cmake. You don't need to specify modules either -- Meson is able to autodetect it in this case.

...

BTW you may use both fmt and spdlog from the Meson WrapDB the same way you do nlohmann_json.

Reply:
This is generated via the Gnu Radio modtool (generating the OOT module build templates). As we'll need changes to that one and apply them to our meson.build files, I'd rather keep this unmodified as much as possible.